phoenix-core/src/main/java/org/apache/phoenix/transaction/TransactionFactory.java

```diff
@@ -22,43 +22,19 @@
 
 public class TransactionFactory {
 
-    private static final PhoenixTransactionProvider tephraTransactionProvider;
-
-    static{
-        boolean tephraEnabled = true;
-        try {
-            Class.forName("org.apache.tephra.TransactionFailureException");
-        } catch (Throwable e) {
-            tephraEnabled = false;
-        }
-        if (tephraEnabled) {
-            tephraTransactionProvider = TephraTransactionProvider.getInstance();
-        } else {
-            tephraTransactionProvider = NotAvailableTransactionProvider.getInstance();
-        }
-    }
-
     public enum Provider {
-        TEPHRA((byte)1, tephraTransactionProvider,
-            tephraTransactionProvider instanceof TephraTransactionProvider),
-        OMID((byte)2, OmidTransactionProvider.getInstance(), true);
+        OMID((byte)2, OmidTransactionProvider.getInstance());
```

Comment:
This changes the ordinal, which SHOULD not be a real problem as we don't use that for serialization, but keeping TEPHRA, and just renaming it to something like TEPHRA_REMOVED would possibly also make it easier to handle the hairy cases where tephra defined for old tables.